gnupg-pkcs11-scd/keyutil.c

```diff
@@ -118,43 +217,87 @@ keyutil_get_cert_mpi (
 		error = GPG_ERR_BAD_CERT;
 		goto cleanup;
 	}
- 
+
 	if ((pubkey = X509_get_pubkey (x509)) == NULL) {
 		error = GPG_ERR_BAD_CERT;
 		goto cleanup;
 	}
- 
-	if ((rsa = EVP_PKEY_get1_RSA(pubkey)) == NULL) {
-		error = GPG_ERR_WRONG_PUBKEY_ALGO;
+
+	pubkey_ctx = EVP_PKEY_CTX_new(pubkey, NULL);
```

Comment:
this is quite new, right? if we add this we should first drop support of older openssl and support only `>=1.1`.

Reply:
https://www.openssl.org/docs/man1.1.1/man3/EVP_PKEY_CTX_new.html says:

> These functions were added in OpenSSL 1.0.0.